### PR TITLE
Add doc_references field to servicelogs

### PR DESCRIPTION
**Pull request by AlexVulaj**
Adds doc_references to servicelogs.

This work supports https://issues.redhat.com/browse/SDE-3653

```
$ ./osdctl servicelog post 28idnvi2ijne88a6nof30spvtvd2lrjv -t https://raw.githubusercontent.com/openshift/managed-notifications/62753642c541bab51c27d5724506201b6ac23352/osd/cluster_stuck.json -p STATE=xyz
...
INFO[0002] The following template will be sent:         
{
  "severity":"Error",
  "service_name":"SREManualAction",
  "summary":"Action required: Cluster Stuck",
  "description":"SRE has observed that your cluster is stuck in an xyz state. If you need assistance to fully remove and cleanup your cluster, please open a support case.",
  "internal_only":false,
  "event_stream_id":"",
  "doc_references": [
    "https://access.redhat.com/support/cases/new"
  ]
}
Continue? (y/N): y
...

$ osdctl servicelog list 28idnvi2ijne88a6nof30spvtvd2lrjv
{
  "items": [
  ...
      {
      "cluster_id":"28idnvi2ijne88a6nof30spvtvd2lrjv",
      "cluster_uuid":"4e5b3876-97ee-46c6-8e9a-813b857758a6",
      "created_at":"2024-02-08T05:04:51.454038Z",
      "created_by":"avulaj",
      "description":"SRE has observed that your cluster is stuck in an xyz state. If you need assistance to fully remove and cleanup your cluster, please open a support case.",
      "doc_references": [
        "https://access.redhat.com/support/cases/new"
      ],
      "event_stream_id":"2c4R6rE85UJD5jNV3wpKPvepk7X",
      "href":"/api/service_logs/v1/cluster_logs/2c4R6nQrVsZo96M6ZWeMYfPIXHw",
      "id":"2c4R6nQrVsZo96M6ZWeMYfPIXHw",
      "internal_only":false,
      "kind":"LogEntry",
      "log_type":"General Notification",
      "service_name":"SREManualAction",
      "severity":"Error",
      "summary":"Action required: Cluster Stuck",
      "timestamp":"2024-02-08T05:04:51.453981Z",
      "username":""
    }
  ],
...
}
```

### CODE REVIEW

**Comment by geowa4** [comment]
We should amend the flow that checks the doc links we send to account for this field. https://github.com/openshift/osdctl/blob/c163a14b0cd86e5795b487718a4d7d2c789473e9/cmd/servicelog/post.go#L283

Do you think we should also pull all doc links from the description to put into this field as a transitional step?

File: internal/servicelog/template.go [hunk 1]
```diff
@@ -54,6 +55,10 @@ func (m *Message) GetSubscriptionID() string {
 	return m.SubscriptionID
 }
 
+func (m *Message) GetDocReferences() []string {
```
Comment (AlexVulaj) [inline internal/servicelog/template.go:58]: Getters aren't really needed for exported types, but I wanted to maintain consistency with the rest of the file.

**Comment by AlexVulaj** [comment]
> We should amend the flow that checks the doc links we send to account for this field.
> ...
> Do you think we should also pull all doc links from the description to put into this field as a transitional step?

As part of https://issues.redhat.com/browse/OSD-19763 , the `doc_references` fields for our SLs were updated after thorough discussion and collaboration between the docs team, MOBB and SREP team. There's a sheet in the OSD card that tracks every link, and those changes can be seen here: https://github.com/openshift/managed-notifications/pull/288

That all said, I don't think osdctl needs to be responsible for any additional logic around the templates.

**Review by typeid — APPROVED** [review]
/lgtm

**Comment by geowa4** [comment]
Good to know that exists, thanks. However, we have that check in osdctl to ensure we don't send the wrong docs for the cluster type. If the description doesn't have a doc link but the doc_references field does, we should check it.

**Pull request merged**